src/commands/deleteCommand.ts

```diff
@@ -0,0 +1,81 @@
+import { setTimeout as wait } from "node:timers/promises";
+import { Command } from "@yuudachi/framework";
+import type { InteractionParam, ArgsParam, CommandMethod } from "@yuudachi/framework/types";
+import { PermissionFlagsBits, time, TimestampStyles } from "discord.js";
+import type { DeleteCommandContextCommand } from "../interactions/context/deleteCommand.js";
+import { GITHUB_THREAD_NAME } from "../util/constants.js";
+
+export default class extends Command<typeof DeleteCommandContextCommand> {
+	public constructor() {
+		super(["Delete Command"]);
+	}
+
+	private matchChannelName(name: string): boolean {
+		return name === GITHUB_THREAD_NAME;
+	}
+
+	public override async messageContext(
+		interaction: InteractionParam<CommandMethod.MessageContext>,
+		args: ArgsParam<typeof DeleteCommandContextCommand>,
+	): Promise<void> {
+		if (args.message.interaction) {
+			if (
+				args.message.interaction.user.id !== interaction.user.id ||
+				!interaction.memberPermissions.has(PermissionFlagsBits.ManageMessages)
+			) {
+				await interaction.reply({ content: "You are not author of this command.", ephemeral: true });
+				return;
+			}
+			
+			if(!args.message.deletable) {
+				await interaction.reply({ content: "I can't delete this message.", ephemeral: true });
+				return
+			} 
+
+			await args.message.delete();
+			await interaction.reply({ content: "Command successfully deleted.", ephemeral: true });
+		} else if (interaction.channel?.isThread()) {
```

Comment:
`interaction.channel`? can this be executed on any message in a GH thread?
i would prob. change that to being executable on the message that started a thread, at most

Reply:
but starting message is used below